include/common/omp.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+// Standard C++ includes
+#include <vector>
+
+namespace BoBRobotics {
+namespace OMP {
+
+//! Returns maximum number of threads for OpenMP or 1 if OpenMP is not in use
+size_t maxThreads();
+
+//! Returns current thread ID or 0 if OpenMP is not in use
+size_t currentThread();
+
+//------------------------------------------------------------------------
+// BoBRobotics::OMP::Array
+//------------------------------------------------------------------------
+/*!
+ * \brief A wrapper providing thread-local storage for e.g. scratch data
+ */
+template<class T>
```

Comment:
I don't quite see why this is necessary as you're using it on static members which can be marked thread local with 
```c++
#pragma omp threadprivate
```
(according to https://www.openmp.org/spec-html/5.0/openmpsu104.html)

Reply:
Good point. I've dropped this code now that I better understand how to use ``threadprivate` :stuck_out_tongue: